src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBulkRecordReader.java

```diff
@@ -123,6 +133,11 @@ public void close() throws IOException {
       // this also closes the inputStream
       csvParser.close();
     }
+    try {
+      SalesforceBulkUtil.closeJob(bulkConnection, jobId);
```

Comment:
Can we just check that the bulkConnection is not null - in case the exception was thrown before or during its instantiation